opentelemetry-sdk/src/opentelemetry/sdk/metrics/__init__.py

```diff
@@ -0,0 +1,241 @@
+    will be discarded for monotonic counter metrics. Counter metrics that
+    have a monotonic option set to False allows negative inputs.
+    """
+
+    HANDLE_TYPE = CounterHandle
+
+    def __init__(
+        self,
+        name: str,
+        description: str,
+        unit: str,
+        value_type: Type[metrics_api.ValueT],
+        label_keys: Sequence[str] = None,
+        enabled: bool = True,
+        monotonic: bool = True,
+    ):
+        super().__init__(
+            name,
+            description,
+            unit,
+            value_type,
+            label_keys=label_keys,
+            enabled=enabled,
+            monotonic=monotonic,
+        )
+
+
+class Gauge(Metric):
+    """See `opentelemetry.metrics.Gauge`.
+
+    By default, gauge values can go both up and down (non-monotonic).
+    Negative inputs will be discarded for monotonic gauge metrics.
+    """
+
+    HANDLE_TYPE = GaugeHandle
+
+    def __init__(
+        self,
+        name: str,
+        description: str,
+        unit: str,
+        value_type: Type[metrics_api.ValueT],
+        label_keys: Sequence[str] = None,
+        enabled: bool = True,
+        monotonic: bool = False,
+    ):
+        super().__init__(
+            name,
+            description,
+            unit,
+            value_type,
+            label_keys=label_keys,
+            enabled=enabled,
+            monotonic=monotonic,
+        )
+
+
+class Measure(Metric):
+    """See `opentelemetry.metrics.Measure`.
+
+    By default, measure metrics can accept both positive and negatives.
+    Negative inputs will be discarded when monotonic is True.
+    """
+
+    HANDLE_TYPE = MeasureHandle
+
+    def __init__(
+        self,
+        name: str,
+        description: str,
+        unit: str,
+        value_type: Type[metrics_api.ValueT],
+        label_keys: Sequence[str] = None,
+        enabled: bool = False,
+        monotonic: bool = False,
+    ):
+        super().__init__(
+            name,
+            description,
+            unit,
+            value_type,
+            label_keys=label_keys,
+            enabled=enabled,
+            monotonic=monotonic,
+        )
+
+
+class Meter(metrics_api.Meter):
+    """See `opentelemetry.metrics.Meter`."""
+
+    def record_batch(
+        self,
+        label_values: Sequence[str],
+        record_tuples: Sequence[Tuple[metrics_api.Metric, metrics_api.ValueT]],
+    ) -> None:
+        """See `opentelemetry.metrics.Meter.record_batch`."""
+        for metric, value in record_tuples:
+            metric.get_handle(label_values).update(value)
+
+    def create_metric(
+        self,
+        name: str,
+        description: str,
+        unit: str,
+        value_type: Type[metrics_api.ValueT],
+        metric_type: Type[metrics_api.MetricT],
+        label_keys: Sequence[str] = None,
+        enabled: bool = True,
+        monotonic: bool = False,
+    ) -> "Metric":
```

Comment:
```suggestion
    ) -> metrics_api.MetricT:
```

https://docs.python.org/3/library/typing.html#generics

And now I remember the reason for ValueT: I thought we should make a MetricT[ValueT] to make mypy detect all cases where we could run into the "wrong value type" check at lint-time.